RelationalRecord/src/Relations/User.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, FlexibleInstances #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module  Relations.User where
+
+import  Types.User as User
+import  Types.DB
+import  Relations.DB
+
+import  Database.Relational.Query
+import  Database.HDBC.Query.TH      (makeRecordPersistableDefault)
+
+import  Data.Int                    (Int32)
+
+
+
+-- SELECTS
+
+allUsers :: Relation () Users
+allUsers = User.users
+
+getUser :: Relation PKey Users
+getUser = relation' . placeholder $ \usrId -> do
+    a       <- query users
+    wheres  $ a ! User.id' .=. usrId
+    return  a
+
+
+-- INSERTS
+
+data UserInsert = UserInsert
+    { iTenantId     :: PKey
+    , iUsername     :: String
+    , iPassword     :: String
+    , iFirstName    :: Maybe String
+    , iLastName     :: Maybe String
+    }
+$(makeRecordPersistableDefault ''UserInsert)
+
+piUser :: Pi Users UserInsert
```

Comment:
HRR derived types have all strict fields; an "identity projection" of such a type would require us to assign a (non-existent) primary key value where present to construct the record -- which will then clash on the DB when we run the insert. For tables without PK, this is absolutely possible; I've done so for the join-through table users_roles, it looks like this:
```
assignRole :: Insert UsersRoles
assignRole = typedInsert (tableOf usersRoles) (defineDirectPi [0, 1])
```